test-apps/display-test-app/src/frontend/SectionDrawingApi.ts

```diff
@@ -0,0 +1,344 @@
+    } else {
+      // View new drawing view state directly (without saving it)
+      await drawingViewState.load();
+      const viewport = IModelApp.viewManager.selectedView;
+      if (!viewport)
+        return;
+      viewport.changeView(drawingViewState);
+    }
+  }
+
+  private static async createSectionDrawingViewStateProps(
+    iModelConnection: IModelConnection,
+    spatialViewDefinitionId: string,
+    sectionDrawingModelId: string,
+    drawingToSpatial: Transform,
+  ): Promise<ViewStateProps> {
+    if (!iModelConnection)
+      return {} as ViewStateProps;
+
+    const dictionaryId = IModelConnection.dictionaryId;
+    // Get all categories
+    const query = "SELECT ECInstanceId from BisCore.Category";
+    const categories = [];
+    for await (const row of iModelConnection.createQueryReader(query, undefined, {
+      rowFormat: QueryRowFormat.UseJsPropertyNames,
+    })) {
+      categories.push(row.id);
+    }
+
+    const spatialToDrawing = drawingToSpatial.inverse()!;
+    const spatialViewState: SpatialViewState = await iModelConnection.views.load(spatialViewDefinitionId) as SpatialViewState;
+    const frustum = spatialViewState.calculateFrustum()!;
+    frustum.multiply(spatialToDrawing);
+    const drawingViewExtents = Range3d.create(frustum.getCorner(Npc.LeftBottomFront), frustum.getCorner(Npc.RightTopFront));
+    
+    // // Visible extents
+    // const drawingViewExtentsProps: Range3dProps =
+    //     await SectionDrawingIpcInvoker.getOrCreate().calculateDrawingViewExtents(
+    //       spatialViewDefinitionId,
+    //       sectionDrawingModelId,
+    //       drawingToSpatial.toJSON(),
+    //     );
+    // const drawingViewExtents = Range3d.fromJSON(drawingViewExtentsProps);
+    const originX = drawingViewExtents.low.x;
+    const originY = drawingViewExtents.low.y;
+    const deltaX = drawingViewExtents.xLength();
+    const deltaY = drawingViewExtents.yLength();
+
+    const categorySelectorProps: CategorySelectorProps = {
+      categories,
+      code: Code.createEmpty(),
+      model: dictionaryId,
+      classFullName: "BisCore:CategorySelector",
+    };
+
+    const viewDefinitionProps: ViewDefinition2dProps = {
+      baseModelId: sectionDrawingModelId,
+      categorySelectorId: "",
+      displayStyleId: "",
+      origin: { x: originX, y: originY },
+      delta: { x: deltaX, y: deltaY },
+      angle: { radians: 0 },
+      code: Code.createEmpty(),
+      model: dictionaryId,
+      classFullName: "BisCore:ViewDefinition2d",
+    };
+
+    const displayStyleProps: DisplayStyleProps = {
+      code: Code.createEmpty(),
+      model: dictionaryId,
+      classFullName: "BisCore:DisplayStyle2d",
+      jsonProperties: {
+        styles: {
+          backgroundColor: ColorDef.white.tbgr,
+        },
+      },
+    };
+
+    // const rotation = spatialViewState.rotation;
+
+    // const transform = Transform.createOriginAndMatrix(spatialViewState.origin, rotation).inverse();
```

Comment:
@Alfonso-Martello it took me far too long to realize you were recomputing an incorrect drawing-to-spatial transform here instead of using the one passed in.

I have fixed  the view extents such that the spatial view will always display in the center of the drawing view. There's still a bit of front plane clipping in some cases, I will circle back on figuring that out in a bit.

Reply:
Thanks @pmconne,

I think we added that early on while trying something out, and then forgot to delete it (thereby messing up all our future attempts).

@Ellord207 and I reviewed your fix and understand what it is doing.

Regarding the plane clipping, there is another .bim file I wanted to share with you as I think it shows back plane clipping occurring as well. I sent you an email titled "Back clipping example .bim" with the .bim file and more details.

Reply:
Was it created using the current version of the code in this PR?

Reply:
Yes